packages/ui-select/src/Select/index.tsx

```diff
@@ -697,10 +697,15 @@ class Select extends Component<SelectProps> {
     // popup buttons rather than comboboxes.
     const overrideProps: Partial<TextInputProps> = !isEditable
       ? {
-          // Given that Safari with Voiceover does not support proper combobox
-          // handling, a button role is set as a workaround.
+          // We need role="combobox" for the 'open list' button shortcut to work
+          // with desktop screenreaders.
+          // But desktop Safari with Voiceover does not support proper combobox
+          // handling, a 'button' role is set as a workaround.
           // See https://bugs.webkit.org/show_bug.cgi?id=236881
-          role: utils.isSafari() ? 'button' : 'combobox',
+          // Also on iOS Chrome with role='combobox' it announces unnecessarily
+          // that its 'read-only' and that this is a 'textfield', see INSTUI-4500
+          role:
+            utils.isSafari() || utils.isAndroidOrIOS() ? 'button' : 'combobox',
```

Comment:
This is the fix. When we use `combobox` role, it doesnt seem to change the role of the trigger element itself, just speaks extra stuff like that it can be opened. If we use the `button` role, it does not announce that its actually an `input`, and neither that its `read-only`. 

On Android sadly its still broken (so no change), I did not use something like `isIOS` because I want to minimize the number of different browser detectors.